spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/web/WebMvcAutoConfiguration.java

```diff
@@ -227,6 +228,15 @@ public LocaleResolver localeResolver() {
 		}
 
 		@Bean
+		@ConditionalOnMissingBean
+		@ConditionalOnProperty(prefix = "spring.mvc", name = "default-locale")
+		public LocaleResolver defaultLocaleResolver() {
```

Comment:
We can may use just one property `spring.mvc.locale` so `FixedLocaleResolver` bean should be loaded by default keeping compatibility in previous projects. But, in other property `spring.mvc.locale-resolver` we can set `org.springframework.web.servlet.i18n.AcceptHeaderLocaleResolver` and then use the value of `spring.mvc.locale` in the `defaultLocale`. This change will allow to people move between the best option without any impact.

WDYT?